runtime/jcl/common/jclcinit.c

```diff
@@ -712,6 +716,38 @@ initializeRequiredClasses(J9VMThread *vmThread, char* dllName)
 		}
 	}
 
+	if (J9_ARE_ANY_BITS_SET(vm->extendedRuntimeFlags2, J9_EXTENDED_RUNTIME2_LOAD_AGENT_MODULE)
+			&& (NULL != vm->modulesPathEntry->extraInfo)) {
+		Trc_JCL_initializeRequiredClasses_addAgentModuleEntry(vmThread);
+		const char *module = vm->jimageIntf->jimagePackageToModule(
+				vm->jimageIntf, (UDATA) vm->modulesPathEntry->extraInfo,
+				"jdk/internal/agent");
+		if (NULL != module) {
+			J9VMSystemProperty *systemProperty = NULL;
+			/* Handle the case where there are no user-specified modules. In this case, there
+			 * is typically one system property set to "ALL-DEFAULT".
+			 */
+			if ((J9SYSPROP_ERROR_NONE == vm->internalVMFunctions->getSystemProperty(vm, ADDMODS_PROPERTY_BASE "0", &systemProperty))
+					&& (0 == strcmp(systemProperty->value, "ALL-DEFAULT"))) {
+				/* this is implicitly an unused property */
+				vm->internalVMFunctions->setSystemProperty(vm, systemProperty, AGENT_MODULE_NAME);
+			} else {
+				char *propNameBuffer = j9mem_allocate_memory(sizeof(ADDMODS_PROPERTY_BASE) + 4, OMRMEM_CATEGORY_VM);
+				if (NULL == propNameBuffer) {
+					Trc_JCL_initializeRequiredClasses_addAgentModuleOutOfMemory(vmThread);
+					return 1;
+				}
+				/* Allow up to 10000 --add-modules arguments. */
+				if (vm->addModulesCount >= 10000) {
```

Comment:
This restriction isn't required, just increase the buffer space being allocated unconditionally to allow the maximum possible.

Reply:
I see a thumbs up and this was resolved, but no change was made.